dango/dex/src/execute/order_creation.rs

```diff
@@ -94,14 +93,53 @@ pub(super) fn create_market_order(
         order.quote_denom
     );
 
-    let deposit = match order.direction {
-        Direction::Bid => Coin {
-            denom: order.quote_denom.clone(),
-            amount: *order.amount,
+    // Load the resting order book of the pair.
+    // The best price available in the book, together with the order's maximum
+    // slippage, will be used to determine the order's "limit price"
+    let resting_order_book = RESTING_ORDER_BOOK
+        .load(storage, (&order.base_denom, &order.quote_denom))
+        .map_err(|err| {
+            anyhow!(
+                "can't created market order, because resting order book either doesn't exist or is corrupted. base denom: {}, quote denom: {}, err: {err}",
+                order.base_denom,
+                order.quote_denom
+            )
+        })?;
+
+    let (price, deposit) = match order.direction {
+        Direction::Bid => {
+            let best_ask_price = resting_order_book.best_ask_price.ok_or_else(|| {
+                anyhow!(
+                    "can't create market bid order, because best ask price isn't avaiable. base denom: {}, quote denom: {}",
+                    order.base_denom,
+                    order.quote_denom
+                )
+            })?;
+
+            let one_add_max_slippage = Udec128_24::ONE.saturating_add(order.max_slippage);
+            let price = best_ask_price.saturating_mul(one_add_max_slippage);
+
+            (price, Coin {
+                denom: order.quote_denom.clone(),
+                amount: order.amount.checked_mul_dec_ceil(best_ask_price)?,
+            })
         },
-        Direction::Ask => Coin {
-            denom: order.base_denom.clone(),
-            amount: *order.amount,
+        Direction::Ask => {
+            let best_bid_price = resting_order_book.best_bid_price.ok_or_else(|| {
+                anyhow!(
+                    "can't create market ask order, because best bid price isn't avaiable. base denom: {}, quote denom: {}",
```

Comment:
```suggestion
                anyhow!(
                    "can't create market ask order, because best bid price isn't available. base denom: {}, quote denom: {}",
```